test/e2e/auth/node_authn.go

```diff
@@ -54,7 +54,7 @@ var _ = SIGDescribe("[Feature:NodeAuthenticator]", func() {
 		gomega.Expect(len(sa.Secrets)).NotTo(gomega.BeZero())
 	})
 
-	ginkgo.It("The kubelet's main port 10250 should reject requests with no credentials", func() {
+	ginkgo.It("The kubelet's main port 10250 should reject requests with no credentials [Config:Kubelet:NoAnonymousAuth]", func() {
```

Comment:
this sounds like a feature of the cluster. we now need to update other test suite configs to skip `[Config:.*]` ?

Reply:
yes, while `Feature:` is a common pattern `Config:` is AFAIK brand new?

Reply:
just checked, it is brand new.

> Enable the Node authentication & restriction E2E tests in most test suites
Though this would indeed add the test to most test suites.
Though one of the issues is that we don't know if this test works (haven't been able to find a place in testgrid where this one is running).So if we added it then a bunch of things may start failing

Reply:
If `pull-kubernetes-e2e-gce` it would be safe (fingers crossed) to merge this, no?

Reply:
> If pull-kubernetes-e2e-gce it would be safe (fingers crossed) to merge this, no?

it may break CI on other deployment tools like kops etc..

Reply:
To clarify, I specifically _don't_ want test suites to skip `[Config:.*]`, which is why I didn't just keep it as a feature. Essentially, I want this to be opt-out rather than opt-in.

I think the real problem is that we don't have a good way of accessing the the cluster configuration in the tests. What I really want to do is say: "if node webhook authentication is disabled, skip this test". Using `[Feature:]` tags doesn't really scale, as we can see from a number of tests that were added but never run.

https://github.com/kubernetes/kubernetes/pull/33706 was another attempt at addressing this.

I also want to make sure the configuration is explicit in the test configuration, rather than scraped from the cluster. If we scraped the configuration from the cluster, we wouldn't catch bugs where a feature was accidentally disabled.

Reply:
not having a good way to determine how the cluster is configured is definitely a problem.
/assign @timothysc @spiffxp

Reply:
Well, some amount of config observability is in our backlog (nobody's working on it right now though). If someone wants to work on making /configz (or some other mechanism) a production ready way to observe live config, please reach out to wg-component-standard: https://github.com/kubernetes/community/tree/master/wg-component-standard.

Even then, using it for dynamic decision making (rather than just monitoring) is not something we've taken a hard look at. Note the e2e tests do have a mechanism to use dynamic config to set up a node for a test. Let me find a code example to link to.

Reply:
*correction: it's a feature of the e2e_node test utils:
- Example usage in cpu_manager_test: https://github.com/kubernetes/kubernetes/blob/ef2016c2530bccff26adf3ca20309e10f4b5b97d/test/e2e_node/cpu_manager_test.go#L178
- Utils: https://github.com/kubernetes/kubernetes/blob/ef2016c2530bccff26adf3ca20309e10f4b5b97d/test/e2e_node/util.go#L180

Reply:
You could probably just port this to the e2e suite if we need it there too.